opal/util/ethtool.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2016 Karol Mroz.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "opal_config.h"
+
+#include <string.h>
+#include <limits.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+#ifdef HAVE_NET_IF_H
+#include <net/if.h>
+#endif
+#ifdef HAVE_LINUX_ETHTOOL_H
+#include <linux/ethtool.h>
+#endif
+#ifdef HAVE_SYS_IOCTL_H
+#include <sys/ioctl.h>
+#endif
+#ifdef HAVE_LINUX_SOCKIOS_H
+#include <linux/sockios.h>
+#endif
+
+#include "opal/util/ethtool.h"
+#include "opal/util/if.h"
+
+#if !defined(HAVE_DECL_ETHTOOL_CMD_SPEED)
+#if !defined(HAVE_STRUCT_ETHTOOL_CMD)
+struct ethtool_cmd;
```

Comment:
This will ultimately result in an error on a system that doesn't have `struct ethtool_cmd`, right?  I.e., you forward reference it here and then use it in the declaration of `ethtool_cmd_speed()` here.  But then I don't see it defined anywhere, even though it is used down on line 67.  I think in this conditional block, you need to declare `struct ethtool_cmd` as well.